docs/deployment/cumulus-distribution.md

```diff
@@ -0,0 +1,37 @@
+---
+id: cumulus_distribution
+title: Using the Cumulus Distribution API
+hide_title: false
+---
+
+The Cumulus Distribution API is a set of endpoints that can be used to enable AWS Cognito authentication when downloading data from S3.
+
+## Configuring a Cumulus Distribution deployment
+
+The Cumulus Distribution API is included in the main [Cumulus](https://github.com/nasa/cumulus/tree/master/tf-modules/cumulus_distribution) repo.
+
+To configure a deployment to use Cumulus Distribution, remove or comment the "Thin Egress App Settings" in [the Cumulus Template Deploy](https://github.com/nasa/cumulus-template-deploy/blob/master/cumulus-tf/main.tf) and enable the Cumulus Distribution settings.
+
+## S3 Bucket Mapping
+
+The Cumulus Distribution API uses a `bucket_map.yaml` or `bucket_map.yaml.tmpl` file to determine which buckets to
+serve. [See the examples](https://github.com/nasa/cumulus/tree/master/example/cumulus-tf/cumulus_distribution).
+
+The default Cumulus module generates a file at `s3://${system_bucket}/distribution_bucket_map.json`.
+
+The configuration file is a simple json mapping of the form:
+
+```json
+{
+  "daac-public-data-bucket": "/path/to/this/kind/of/data"
+}
+```
+
+> Note: Cumulus only supports a one-to-one mapping of bucket->Cumulus Distribution path for 'distribution' buckets.
+
+> Note: The bucket map **must include mappings for all of the `protected` and `public` buckets specified in the `buckets` variable in `cumulus-tf/terraform.tfvars`**, otherwise Cumulus may not be able to determine the correct distribution URL for ingested files and you may encounter errors.
+
+## Switching from the Thin Egress App to Cumulus Distribution
+
```

Comment:
Ah ok, right. We don't include cumulus_distribution in the template deploy yet. I will make a new `cumulus_distribution.tf.example` and a directory for the bucket maps: `cumulus-tf/cumulus_distribution`.

Reply:
I added the module in the template deploy repo here:

https://github.com/nasa/cumulus-template-deploy/pull/63/files

and updated these docs here:

77209a99bebb9ab34c393a612717e4c6724cd3bb

I didn't call out the bucket mapping explicitly because we have it in the above `S3 Bucket Mapping` section but is there something I should highlight?

This is deploying now so is currently untested. I'll make sure this actually does what I think it does.

Reply:
ok.  Also need to add cognito specific variables to `cumulus-template-deploy` and instructions so the deployment for cognito would work.

Reply:
Yup, you're totally right. I'm adding these things to this repo but haven't tested a full deploy using Cognito from it. I'll commit once it's fully deployed and working.

Reply:
Here's an update to the docs here and I'll add the variable and test in cumulus-template-deploy

https://github.com/nasa/cumulus/pull/2353/commits/1d49b58df622bb0df9d30ef68c7637ab1abcc1a4

Reply:
Also the csdap_* variables https://github.com/nasa/cumulus/blob/master/example/cumulus-tf/terraform.tfvars.example#L64